File: src/main/java/com/github/anba/es6draft/ast/scope/Name.java
/**
 * Copyright (c) André Bargull
 * Alle Rechte vorbehalten / All Rights Reserved.  Use is subject to license terms.
 *
 * <https://github.com/anba/es6draft>
 */
package com.github.anba.es6draft.ast.scope;

/**
 * 
 */
public final class Name implements Cloneable {
    public static final String DEFAULT_EXPORT = "*default*";
    private final String identifier;

    /**
     * Constructs a new Name object.
     * 
     * @param identifier
     *            the identifier string
     */
    public Name(String identifier) {
        this.identifier = identifier;
    }

    /**
     * Returns the identifier string.
     * 
     * @return the identifier string
     */
    public String getIdentifier() {
        return identifier;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != Name.class) {
            return false;
        }
        return identifier.equals(((Name) obj).identifier);
    }

    @Override
    public int hashCode() {
        return identifier.hashCode();
    }

    @Override
    public String toString() {
        return identifier;
    }

    @Override
    public Name clone() {
        return new Name(identifier);
    }
}

File: src/test/scripts/suite262/intl402/Intl/getCanonicalLocales/name.js
/*
 * Copyright (c) André Bargull
 * Alle Rechte vorbehalten / All Rights Reserved.  Use is subject to license terms.
 *
 * <https://github.com/anba/es6draft>
 */
/*---
esid: sec-intl.getcanonicallocales
description: >
  Intl.getCanonicalLocales.name is "getCanonicalLocales".
info: >
  8.2.1 Intl.getCanonicalLocales (locales)

  17 ECMAScript Standard Built-in Objects:
    Every built-in Function object, including constructors, that is not
    identified as an anonymous function has a name property whose value
    is a String. Unless otherwise specified, this value is the name that
    is given to the function in this specification. For functions that
    are specified as properties of objects, the name value is the property
    name string used to access the function.

    Unless otherwise specified, the name property of a built-in Function
    object, if it exists, has the attributes { [[Writable]]: false,
    [[Enumerable]]: false, [[Configurable]]: true }.
includes: [propertyHelper.js]
---*/

assert.sameValue(Intl.getCanonicalLocales.name, "getCanonicalLocales");

verifyNotEnumerable(Intl.getCanonicalLocales, "name");
verifyNotWritable(Intl.getCanonicalLocales, "name");
verifyConfigurable(Intl.getCanonicalLocales, "name");
